code/modules/client/client_procs.dm

```diff
@@ -200,6 +200,11 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 	if(CONFIG_GET(flag/respect_upstream_bans) || CONFIG_GET(flag/respect_upstream_permabans))
 		check_upstream_bans()
 
+	data = GLOB.client_data[ckey]
+	if(!data)
+		data = new(ckey)
+		GLOB.client_data[ckey] = data
```

Comment:
Modify the client data list inside of the client data datum itself instead of here

Reply:
that makes the existing datum unreachable unless you make a new `/datum/client_data` (or a dummy one)